--- VMware-player-16.1.2.17966106/install.sh.orig	2021-10-02 16:18:51.036146585 +0200
+++ VMware-player-16.1.2.17966106/install.sh	2021-10-02 16:19:43.259477667 +0200
@@ -18,7 +18,7 @@ if [ -z "$BASH" ]; then
    fi
 
    # dash flips out of $opts is quoted, so don't.
-   exec /usr/bin/env bash $opts "$SOURCE" "$@"
+   exec /usr/bin/env bash $opts "$0" "$@"
    echo "Unable to restart with bash shell"
    exit 1
 fi
@@ -85,7 +85,7 @@ on_exit() {
    fi
 }
 
-trap on_exit EXIT
+#trap on_exit EXIT
 trap "" USR1
 
 # Retrives and sets the various lengths that are extracted from the
@@ -182,7 +182,7 @@ is_relative() {
 # Returns:
 #    None
 extract_self() {
-   VMIS_TEMP=`mktemp -d /tmp/vmis.XXXXXX`
+   VMIS_TEMP=$(pwd)
    local file="$SOURCE"
    local filter=""
 
@@ -206,7 +206,7 @@ extract_self() {
 }
 
 extract_prepayload() {
-   PREPAYLOAD=`mktemp -d /tmp/vmis.XXXXXX`
+   PREPAYLOAD=$(pwd)
    local file="$SOURCE"
 
    if [ ! -d "$PREPAYLOAD" ]; then
@@ -575,6 +575,7 @@ main() {
       esac
    fi
 
+   if [ ! -d install ]; then
    if ! set_lengths "$SOURCE"; then
       echo "Unable to extract lengths from bundle."
       exit 1
@@ -582,6 +583,9 @@ main() {
 
    extract_prepayload
    extract_self
+   else
+	   VMIS_TEMP=$(pwd)
+   fi
 
    install "$VMIS_TEMP" "$help" "$fullpath" "$extract" "$@"
 }
