diff -Nur old/vmware-any-any-update117d/vmmon-only/Makefile new/vmware-any-any-update117d/vmmon-only/Makefile
--- old/vmware-any-any-update117d/vmmon-only/Makefile	2008-08-30 15:29:13.946349500 +0000
+++ new/vmware-any-any-update117d/vmmon-only/Makefile	2008-08-30 15:29:23.042918000 +0000
@@ -14,9 +14,9 @@
 VM_UNAME = $(shell uname -r)
 
 # Header directory for the running kernel
-HEADER_DIR = /lib/modules/$(VM_UNAME)/build/include
+HEADER_DIR = /usr/src/linux/include
 
-BUILD_DIR = $(HEADER_DIR)/..
+BUILD_DIR = .
 
 DRIVER := vmmon
 
@@ -55,9 +55,6 @@
 
 VMCCVER := $(shell $(CC) -dumpversion)
 
-# If there is no version defined, we are in toplevel pass, not yet in kernel makefiles...
-ifeq ($(VERSION),)
-
 ifeq ($(VM_KBUILD), 24)
 DRIVER_KO := $(DRIVER).o
 else
@@ -113,17 +110,6 @@
 
 include $(SRCROOT)/Makefile.kernel
 
-ifdef TOPDIR
-ifeq ($(VM_KBUILD), 24)
-
-O_TARGET := $(DRIVER).o
-
-obj-y := $($(DRIVER)-y)
-
-include $(TOPDIR)/Rules.make
-endif
-endif
-
 else
 
 include $(SRCROOT)/Makefile.normal
diff -Nur old/vmware-any-any-update117d/vmnet-only/Makefile new/vmware-any-any-update117d/vmnet-only/Makefile
--- old/vmware-any-any-update117d/vmnet-only/Makefile	2008-08-30 15:30:05.813591000 +0000
+++ new/vmware-any-any-update117d/vmnet-only/Makefile	2008-08-30 15:30:13.886095500 +0000
@@ -14,9 +14,9 @@
 VM_UNAME = $(shell uname -r)
 
 # Header directory for the running kernel
-HEADER_DIR = /lib/modules/$(VM_UNAME)/build/include
+HEADER_DIR = /usr/src/linux/include
 
-BUILD_DIR = $(HEADER_DIR)/..
+BUILD_DIR = .
 
 DRIVER := vmnet
 
@@ -55,9 +55,6 @@
 
 VMCCVER := $(shell $(CC) -dumpversion)
 
-# If there is no version defined, we are in toplevel pass, not yet in kernel makefiles...
-ifeq ($(VERSION),)
-
 ifeq ($(VM_KBUILD), 24)
 DRIVER_KO := $(DRIVER).o
 else
